src/shared/logs-show.c

```diff
@@ -368,72 +368,68 @@ static int output_timestamp_realtime(
                 usec_t usec) {
 
         char buf[CONST_MAX(FORMAT_TIMESTAMP_MAX, 64U)];
+        int r;
 
         assert(f);
         assert(j);
 
         if (!VALID_REALTIME(usec))
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "No valid realtime timestamp available.");
 
-        if (IN_SET(mode, OUTPUT_SHORT_FULL, OUTPUT_WITH_UNIT)) {
-                const char *k;
+        switch (mode) {
 
-                if (flags & OUTPUT_UTC)
-                        k = format_timestamp_style(buf, sizeof(buf), usec, TIMESTAMP_UTC);
-                else
-                        k = format_timestamp(buf, sizeof(buf), usec);
-                if (!k)
+        case OUTPUT_SHORT_FULL:
+        case OUTPUT_WITH_UNIT: {
+                if (!format_timestamp_style(buf, sizeof(buf), usec, flags & OUTPUT_UTC ? TIMESTAMP_UTC : TIMESTAMP_PRETTY))
                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                                "Failed to format timestamp: %" PRIu64, usec);
+                break;
+        }
 
-        } else {
+        case OUTPUT_SHORT_UNIX:
+                xsprintf(buf, "%10" PRI_USEC ".%06" PRI_USEC, usec / USEC_PER_SEC, usec % USEC_PER_SEC);
+                break;
+
+        case OUTPUT_SHORT:
+        case OUTPUT_SHORT_PRECISE:
+        case OUTPUT_SHORT_ISO:
+        case OUTPUT_SHORT_ISO_PRECISE: {
                 struct tm tm;
-                time_t t;
 
-                t = (time_t) (usec / USEC_PER_SEC);
+                r = localtime_or_gmtime_usec(usec, flags & OUTPUT_UTC, &tm);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to convert timestamp to calendar time: %m");
```

Comment:
Previously, such error was ignored. But, now correctly handled. So, the fuzzer for journal-remote triggers assertion on i386:
```
==259==ERROR: AddressSanitizer: ABRT on unknown address 0x00000103 (pc 0xf0ca5579 bp 0xffa21e1c sp 0xffa21e00 T0)
SCARINESS: 10 (signal)
/usr/local/bin/llvm-symbolizer: error: 'linux-gate.so.1': No such file or directory
    #0 0xf0ca5579  (linux-gate.so.1+0x579) (BuildId: ba6af82cf50c9cd191788af0425793b9aaa29de2)
    #1 0xefd33415 in raise (build-out/src/shared/libc.so.6+0x31415)
    #2 0xefd1b3f6 in abort (build-out/src/shared/libc.so.6+0x193f6)
    #3 0xf04dc6da in log_assert_failed /work/build/../../src/systemd/src/basic/log.c:996:9
    #4 0x60d77192 in LLVMFuzzerTestOneInput /work/build/../../src/systemd/src/journal-remote/fuzz-journal-remote.c
    #5 0x60d9b59d in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned int) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:614:13
    #6 0x60d9acf8 in fuzzer::Fuzzer::RunOne(unsigned char const*, unsigned int, bool, fuzzer::InputInfo*, bool, bool*) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:516:7
    #7 0x60d9cd7f in fuzzer::Fuzzer::ReadAndExecuteSeedCorpora(std::__Fuzzer::vector<fuzzer::SizedFile, std::__Fuzzer::allocator<fuzzer::SizedFile>>&) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:829:7
    #8 0x60d9d0bd in fuzzer::Fuzzer::Loop(std::__Fuzzer::vector<fuzzer::SizedFile, std::__Fuzzer::allocator<fuzzer::SizedFile>>&) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:867:3
    #9 0x60d8b106 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned int)) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerDriver.cpp:914:6
    #10 0x60db7957 in main /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerMain.cpp:20:10
    #11 0xefd1ced4 in __libc_start_main (build-out/src/shared/libc.so.6+0x1aed4)
    #12 0x60c8ac84 in _start (build-out/fuzz-journal-remote+0x26c84)
```

Please drop the error code check below (or, at least ignore -EINVAL and -ERANGE):
https://github.com/systemd/systemd/blob/f7f062bf9f194dc14a4ffc6e7fff9d8bfdea2147/src/journal-remote/fuzz-journal-remote.c#L87-L88

Reply:
I reworked this logic quite a bit now. We'll now output a fallback timestamp if time_t is too small. That's what we do in the regualr format_timestamp() case as well, after all.